File: moviper/src/main/java/com/mateuszkoslacz/moviper/base/interactor/BaseRxInteractor.java
package com.mateuszkoslacz.moviper.base.interactor;


import com.mateuszkoslacz.moviper.iface.interactor.ViperInteractor;
import com.mateuszkoslacz.moviper.iface.interactor.ViperRxInteractor;

/**
 * Created by lucas.urbas on 29/08/15.
 * <p>
 * Adapted and modified by mateuszkoslacz on 21.10.2016.
 * <p>
 * Base Interactor class. (see {@link ViperInteractor})
 * <p>
 * It's responsible for manipulating data on behalf of presenter, ie. saving data to db, getting
 * data from remotes etc.
 * <p>
 * It has no Presenter reference as it should return Observables to asynchronously pass data to
 * Presenter. If you are looking for solution adopted to regular, non-Rx approach, see {@link
 * BaseRxInteractor}.
 */
public abstract class BaseRxInteractor implements ViperRxInteractor {

    @Override
    public void attach() {

    }

    @Override
    public void detach(boolean retainInstance) {

    }
}

File: moviper3/src/main/java/com/mateuszkoslacz/moviper3/base/interactor/BaseRxInteractor.kt
package com.mateuszkoslacz.moviper3.base.interactor


import com.mateuszkoslacz.moviper3.iface.interactor.ViperRxInteractor
import io.reactivex.disposables.CompositeDisposable

/**
 * Created by lucas.urbas on 29/08/15.
 *
 *
 * Adapted and modified by mateuszkoslacz on 21.10.2016.
 *
 *
 * Base Interactor class. (see ViperInteractor)
 *
 *
 * It's responsible for manipulating data on behalf of presenter, ie. saving data to db, getting
 * data from remotes etc.
 *
 *
 * It has no Presenter reference as it should return Observables to asynchronously pass data to
 * Presenter.
 */
abstract class BaseRxInteractor : ViperRxInteractor {

    private val disposables = CompositeDisposable()

    override fun attach() = Unit

    override fun detach(retainInstance: Boolean) = Unit

    override fun detach() = detach(retainInstance = true)

    override fun destroy() {
        detach(false)
        disposables.dispose()
    }
}
